include/nbl/video/IDeviceMemoryBacked.h

```diff
@@ -13,12 +13,38 @@ namespace nbl::video
 struct NBL_API2 ICleanup
 {
     virtual ~ICleanup() = 0;
+
+    std::unique_ptr<ICleanup> next;
+
+    static void chain(std::unique_ptr<ICleanup>& first, std::unique_ptr<ICleanup>&& next)
+    {
+        if (first)
+            return chain(first->next, std::move(next));
+        first = std::move(next);
+    }
```

Comment:
don't like it being the default, if you want to chain up cleanups the proper way would be to either:
- have a "metacleanup" that inherits from multiple cleanups (and each of them virtually from ICleanup)
- have this same chain, but in a `CLinkedListCleanup final`

Anyway this is probably not needed for anything due to what I think is the misuse of the API https://github.com/Devsh-Graphics-Programming/Nabla/pull/517/files#r1264588641